fix: TCK harness integer properties compared as strings (~423 TCK failures)

## Problem

The TCK test harness (`tests/tck/conftest.py`) incorrectly coerces integer property values to strings when building expected result rows from Gherkin step tables.

When a TCK scenario specifies a node with `{num: 1}`, the conftest `_parse_value` / `_parse_map_literal` pipeline serialises the expected value as `'1'` (string) instead of `1` (integer). The engine correctly returns integer `1`, so the comparison fails.

**Example failure:**
\`\`\`
AssertionError: Expected row not found: {'a': {'labels': [], 'properties': {'num': '1', 'text': 'b'}}}
assert {'a': {'labels': [], 'properties': {'num': '1', 'text': 'b'}}} in
       [{'a': {'labels': [], 'properties': {'num': 1,   'text': 'b'}}}]
\`\`\`

## Scope

~423 failing TCK scenarios. Affects any test that compares node/relationship properties containing numeric values.

## Root cause

`_parse_value` in `tests/tck/conftest.py` is not converting bare numeric strings to integers/floats before inserting them into the expected results dict. Related to the `_parse_map_literal` key quote-stripping fix in #247 but applies to *values*, not keys.

## Fix

Audit `_parse_value` to correctly convert:
- Unquoted integers → `int`
- Unquoted floats → `float`
- Quoted strings → `str` (stripping surrounding quotes)
- `true`/`false` → `bool`
- `null` → `None`

## Related

- #247 (map key quote-stripping fix — companion to this issue)
- Closed: #180 (TCK failure investigation)

--- a/tests/tck/conftest.py
+++ b/tests/tck/conftest.py
@@ -592,12 +592,13 @@ def _parse_node_pattern(pattern: str) -> dict:
                 if ":" in pair:
                     key, val = pair.split(":", 1)
                     key = key.strip()
-                    val = val.strip()
-                    # Remove quotes from string values
-                    if val.startswith("'") and val.endswith("'"):
-                        properties[key] = val[1:-1]
+                    parsed = _parse_value(val.strip())
+                    if hasattr(parsed, "value"):
+                        properties[key] = parsed.value
+                    elif isinstance(parsed, CypherNull):
+                        properties[key] = None
                     else:
-                        properties[key] = val
+                        properties[key] = parsed
 
     return {"labels": sorted(labels), "properties": properties}
 

